packages/Main/src/Layer/OGC3DTilesLayer.js

```diff
@@ -335,20 +333,58 @@ class OGC3DTilesLayer extends GeometryLayer {
      */
     _setupCacheAndQueues(view) {
         const id = view.id;
+        // Share the caches and queues to cut down on memory and correctly prioritize downloads.
+        // https://github.com/NASA-AMMOS/3DTilesRendererJS?tab=readme-ov-file#multiple-tilesrenderers-with-shared-caches-and-queues
         if (viewers[id]) {
+
             this.tilesRenderer.lruCache = viewers[id].lruCache;
             this.tilesRenderer.downloadQueue = viewers[id].downloadQueue;
             this.tilesRenderer.parseQueue = viewers[id].parseQueue;
+
+            // Store references to all layer callbacks for this view
+            // Each tile has it's own callback
+            if (!viewers[id].layerCallbacks) {
+                viewers[id].layerCallbacks = [];
+            }
+            viewers[id].layerCallbacks.push(this._originalSchedulingCallback);
+
+            // Create a combined callback that calls all layer callbacks
+            const combinedCallback = (func) => {
+                if (viewers[id] && viewers[id].layerCallbacks) {
+                    // Filter out null/undefined callbacks (from deleted layers)
+                    viewers[id].layerCallbacks = viewers[id].layerCallbacks.filter(callback => callback != null);
+                    viewers[id].layerCallbacks.forEach((callback) => {
+                        if (callback) { callback(func); }
+                    });
+                }
+            };
+
+            // Set the combined callback
+            // We set our scheduling callback for tiles downloading and parsing -> MANDATORY for VR
+            // (WebXR session has its own requestAnimationFrame method separate from that of the window
+            //  https://github.com/NASA-AMMOS/3DTilesRendererJS/issues/213#issuecomment-947943386)
+            // Necessary to update rendering of the tiles in VR
+            // Example: https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/de25d27dc0e75278962b5f401faee30f8dce2fe0/example/vr.js
+            this.tilesRenderer.downloadQueue.schedulingCallback = combinedCallback;
+            this.tilesRenderer.parseQueue.schedulingCallback = combinedCallback;
         } else {
+            // Set the combined callback (cf. above comment)
+            this.tilesRenderer.downloadQueue.schedulingCallback = this.tilesSchedulingCB;
+            this.tilesRenderer.parseQueue.schedulingCallback = this.tilesSchedulingCB;
+
             viewers[id] = {
                 lruCache: this.tilesRenderer.lruCache,
                 downloadQueue: this.tilesRenderer.downloadQueue,
                 parseQueue: this.tilesRenderer.parseQueue,
+                layerCallbacks: [this._originalSchedulingCallback],
```

Comment:
IMHO, you should use a map from layer ids to callbacks (or an object). It should simplify the code as you will not have to null-check.

I think of something like:

```suggestion
                layerCallbacks: {
                   [this.id]: this._originalSchedulingCallback,
                },
```